api/use_cases/get_users_count.py

```diff
@@ -0,0 +1,35 @@
+from api.use_cases import UseCase
+from api.share.requests.users import GetUsersCountRequest
+from api.models.people import PersonalInfo
+from django.db.models.functions import Lower
+
+
+class UsersCount(UseCase):
+
+    def __int__(self):
+        pass
+
+    def process_request(self, request_object: GetUsersCountRequest):
+        # business logic
+        registration_bot_chat_id = request_object.registration_bot_chat_id
+        tg_username = request_object.tg_username
+        phone = request_object.phone
+        email = request_object.email
+        first_name = request_object.first_name
+        last_name = request_object.last_name
+
+        res = PersonalInfo.objects
+        if registration_bot_chat_id:
+            res = res.filter(registration_bot_chat_id=registration_bot_chat_id)
+        if tg_username:
+            res = res.filter(tg_username__iexact=tg_username)
+        if phone:
+            res = res.filter(phone=phone)
+        if email:
+            res = res.filter(email__iexact=email)
+        if first_name:
+            res = res.filter(first_name__iexact=first_name)
+        if last_name:
+            res = res.filter(last_name__iexact=last_name)
```

Comment:
it looks like this workflow will lead to too many database queries. Or are they queried once (all of them are SELECTed) and then stored in memory by the time you run `filter`? In this case, when the amount of `PersonalInfo` object gets into tens of thousands, we may be in a little trouble?

Reply:
As a understand, django's queries are lazy. "QuerySets are lazy – the act of creating a QuerySet doesn’t involve any database activity. You can stack filters together all day long, and Django won’t actually run the query until the QuerySet is evaluated." In this case SQL query evaluated in res.count() and it looks like "SELECT ..... WHERE () AND () AND () ...."

Reply:
OK great!